### DISCUSSION TITLE
How do you enabled breaking change check in AL-GO

### DISCUSSION THREAD

**Discussion by gntpet** [microsoft/AL-Go#1583] [Ideas]
Hey @freddydk, 

Do you have some docs about 'braking change' detection. I see that our pipeline downloads previous release app. But, I don't see any error or warning regarding breaking changes. Is it mandatory to use container for breaking change detection?

Our configuration is like this:

```
{
  "type": "PTE",
  "templateUrl": "https://github.com/microsoft/AL-Go-PTE@main",
  "treatTestFailuresAsWarnings": true,
  "doNotRunBcptTests": true,
  "enableCodeCop": true,
  "rulesetFile": "../al.ruleset.json",
  "companyName": "TNM Automated Test",
  "runs-on": "srs-eid",
  "githubRunner": "srs-eid",
  "environments": [
    "QA"
  ],
  "CICDPushBranches": [
    "main",
    "release/*"
  ],
  "CICDPullRequestBranches": [
    "main",
    "release/*",
    "features/*",
    "bug/*"
  ],
  "DeployToQA": {
    "EnvironmentType": "SaaS",
    "EnvironmentName": "QA",
    "Projects": "SRS.nl,SRS.base",
    "Branches": [
      "main"
    ],
    "ContinuousDeployment": true,
    "runs-on": "srs-eid"
  },
  "DeployToUAT": {
    "EnvironmentType": "SaaS",
    "EnvironmentName": "UAT",
    "Projects": "SRS.nl,SRS.base",
    "Branches": [
      "release/*"
    ],
    "ContinuousDeployment": false,
    "runs-on": "srs-eid"
  },
  "DeployToPROD": {
    "EnvironmentType": "SaaS",
    "EnvironmentName": "PROD",
    "Projects": "SRS.nl,SRS.base",
    "Branches": [
      "release/*"
    ],
    "ContinuousDeployment": false,
    "runs-on": "srs-eid"
  },
  "EnableTaskScheduler": true,
  "doNotRunTests": true,
  "useCompilerFolder": true,
  "doNotPublishApps": true,
  "alwaysBuildAllProjects": true,
  "PullRequestTrigger": "pull_request",
  "templateSha": "1e6ebb58fc629cc134759f996f9be9ab6ada7fb4"
}
```

Best Regards, 
Gintautas

**Comment by mazhelez** [comment]
Hi @gntpet and thank you for your question.

Currently, AL-Go and _BCContainerHelper_ do not support running breaking changes validation. It could be an enhancement we can consider for future versions.

You can check how we did it in BCApps, using a `PreCompileApp.ps1` override.
Key function is [Enable-BreakingChangesCheck](https://github.com/microsoft/BCApps/blob/855e1bad641b2ef045878668099519711e151fc8/build/scripts/GuardingV2ExtensionsHelper.psm1#L15) that creates the `AppSourceCop.json`, that is needed to trigger the breaking changes checks when compiling the app. See more info about it [here](https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/analyzers/appsourcecop#configuration).

Let me know if you have more questions.

**Comment by freddydk** [comment]
Note that the AppSourceCop.json is automatically created if the repo is an AppSource app, with breaking change validation.

**Comment by bcapps-co** [comment]
I have an app that is still in the pre-release stages and as a result I want breaking changes to be allowed. Al-go is failing to build due to breaking changes. How can I ignore this to get the build to run?

**Comment by bcapps-co** [comment]
> I have an app that is still in the pre-release stages and as a result I want breaking changes to be allowed. Al-go is failing to build due to breaking changes. How can I ignore this to get the build to run?

OK, I've figured this out. I added "skipUpgrade" : true to .AL-Go/settings.json and it worked. Yay!
